src/Lib/Canvas.vala

```diff
@@ -73,7 +72,49 @@ public class Akira.Lib.Canvas : Goo.Canvas {
 
     public Canvas (Akira.Window window) {
         Object (window: window);
+        // Make The canvas the drop target
+        Gtk.drag_dest_set (this, Gtk.DestDefaults.ALL, targets, Gdk.DragAction.COPY);
+        this.drag_data_received.connect(this.on_drag_data_received);
     }
+    // Making The canvas recive the dropped file
+    private const Gtk.TargetEntry[] targets = {{"text/uri-list",0,0}};
+    private void on_drag_data_received (Gdk.DragContext drag_context, int x, int y, 
+        Gtk.SelectionData data, uint info, uint time)
+    {
+    //loop through list of Files
+    const string[] ACCEPTED_TYPES = {
+        "jpeg",
+        "png",
+        "tiff",
+        "svg",
+        "xml",
+        "gif"
+    };
+    foreach(string uri in data.get_uris ()){
+        string file = uri.replace("file://","").replace("file:/","");
+        file = Uri.unescape_string (file);
+        // Creating a Glib.File using a file path
+        var img_File = GLib.File.new_for_path(file);
+        var file_img_mangager = new Akira.Lib.Managers.ImageManager(img_File,GLib.Random.int_range(0,100));
+        var root = window.main_window.main_canvas.canvas.get_root_item();
+        var img_extention = Utils.Image.get_extension(img_File);
+        // See if File is an Image File
+        foreach(string accepted_type in ACCEPTED_TYPES){
+            if(img_extention == accepted_type){
+                // Tell Items Manager that we want to insert an Image
+                window.items_manager.insert_image(file_img_mangager);
+                // Insert Image to Canvas
+                window.items_manager.insert_item(x,y,file_img_mangager,artboards[0]);
+                // Requesting Updates from the Root
+                this.request_update();
+                root.request_update();
```

Comment:
These are wrong.
If you use the methods we already have to add images and layers, you shouldn't need to request the update.